From 545436f96af324fc476ff44434da77a6629b7b36 Mon Sep 17 00:00:00 2001
From: Davis Goodin <dagood@microsoft.com>
Date: Mon, 23 Nov 2020 18:42:37 -0600
Subject: [PATCH 14/14] Make Microsoft.Win32.SystemEvents explicit ref

Change Microsoft.Win32.SystemEvents from
_TransitiveExternalAspNetCoreAppReference to
ExternalAspNetCoreAppReference. Normally SystemEvents is brought in by
System.Drawing.Common as a transitive ref, but in source-build,
System.Drawing.Common doesn't have this reference. Ideally this will be
fixed in dotnet/runtime, but for now this is a more straightforward
place to patch the reference.

Without this, the ASP.NET Core shared framework ends up missing the
SystemEvents assembly, making it potentially unable to run some
applications built using the Microsoft-built SDK.
---
 eng/SharedFramework.External.props | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/eng/SharedFramework.External.props b/eng/SharedFramework.External.props
index a43f82e803..246678e723 100644
--- a/eng/SharedFramework.External.props
+++ b/eng/SharedFramework.External.props
@@ -40,6 +40,7 @@
     <ExternalAspNetCoreAppReference Include="Microsoft.Extensions.Options.DataAnnotations"                 Version="$(MicrosoftExtensionsOptionsDataAnnotationsVersion)" />
     <ExternalAspNetCoreAppReference Include="Microsoft.Extensions.Options"                                 Version="$(MicrosoftExtensionsOptionsVersion)" />
     <ExternalAspNetCoreAppReference Include="Microsoft.Extensions.Primitives"                              Version="$(MicrosoftExtensionsPrimitivesVersion)" />
+    <ExternalAspNetCoreAppReference Include="Microsoft.Win32.SystemEvents"                                 Version="$(MicrosoftWin32SystemEventsVersion)" />
     <ExternalAspNetCoreAppReference Include="System.IO.Pipelines"                                          Version="$(SystemIOPipelinesVersion)" />
     <ExternalAspNetCoreAppReference Include="System.Security.Cryptography.Xml"                             Version="$(SystemSecurityCryptographyXmlVersion)" />
 
@@ -52,7 +53,6 @@
 
       If these are needed as direct dependencies, it is okay to change them to ExternalAspNetCoreAppReference and move up into sections above.
     -->
-    <_TransitiveExternalAspNetCoreAppReference Include="Microsoft.Win32.SystemEvents"                       Version="$(MicrosoftWin32SystemEventsVersion)" />
     <_TransitiveExternalAspNetCoreAppReference Include="System.Drawing.Common"                              Version="$(SystemDrawingCommonVersion)" />
     <_TransitiveExternalAspNetCoreAppReference Include="System.Security.Cryptography.Pkcs"                  Version="$(SystemSecurityCryptographyPkcsVersion)" />
     <!--
-- 
2.18.0

